roles/nodejs/tasks/main.yml

```diff
@@ -57,54 +24,53 @@
   register: yarnmit_servers
   ignore_errors: true
 
-- name: Add upstream Yarn APT key from PGP
+- name: nodejs | Add upstream Yarn APT key from PGP
   apt_key:
     id: "{{ nodejs__yarn_upstream_key_id }}"
     state: present
     keyserver: "keyserver.pgp.com"
   when: yarnmit_servers is failed
   ignore_errors: true
 
-- name: Configure upstream APT repository
-  apt_repository:
-    repo: "{{ nodejs__upstream_repository }}"
-    state: present
-    update_cache: true
-  register: nodejs_repo
-  retries: 3
-  delay: 60
-  until: nodejs_repo is succeeded
-
-- name: Add upstream Yarn APT repository
+- name: nodejs | Add upstream Yarn APT repository
   apt_repository:
     repo: "{{ nodejs__yarn_upstream_repository }}"
     state: present
     update_cache: true
   ignore_errors: true
 
-- name: Gather the apt package facts
-  package_facts:
-    manager: auto
-
-- name: Get the versions of nodejs installed
-  set_fact:
-    nodejs_versions: "{{ packages['nodejs'] | map(attribute='version') | list }}"
-  when: "'nodejs' in packages"
+- name: nodejs | uninstall any apt installed node
+  ansible.builtin.apt:
+    name: ["nodejs"]
+    state: absent
+  changed_when: false
 
-- name: Get the version we are trying to install into a regex
-  set_fact:
-    nodejs_wanted_version: "{{ nodejs__upstream_release | regex_replace('node_') | regex_replace('.x') | regex_replace('^(.*)$', '^\\1.') }}"
+- name: nodejs | download nodejs version
+  ansible.builtin.unarchive:
+    src: "{{ nodejs_release_url }}/{{ desired_version }}/node-{{ desired_version }}-linux-x64.tar.gz"
+    dest: /usr/local
+    remote_src: true
+  changed_when: false
 
-- name: Desired node version match regex
-  debug:
-    msg: "{{ nodejs_wanted_version }}"
+- name: nodejs | create symbolic link for node
+  ansible.builtin.file:
+    src: "/usr/local/node-{{ desired_version }}-linux-x64/bin/node"
+    dest: "/usr/local/bin/node"
+    owner: root
+    group: root
+    state: link
+  changed_when: false
```

Comment:
I assume `changed_when: false` is here to make idempotence pass . . .

Reply:
yep